src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -627,4 +630,111 @@ private function fallBackToForUpdate(QueryBuilder $query, string $sql): string
             return $sql;
         }
     }
+
+    private function getMessageForMySQLPlatform(): ?array
+    {
+        $possibleIdsToClaim = $this->createAvailableMessagesQueryBuilder()
+            ->select('id')
+            ->orderBy('available_at', 'ASC')
+            ->setMaxResults(100)->fetchFirstColumn();
+
+        if (0 === \count($possibleIdsToClaim)) {
+            $this->queueEmptiedAt = microtime(true) * 1000;
+
+            return null;
+        }
+        $this->queueEmptiedAt = null;
+
+        foreach ($possibleIdsToClaim as $id) {
+            $claimedId = $this->claimMessage($id);
+            if (null !== $claimedId) {
+                break;
+            }
+        }
+
+        if (!isset($claimedId)) {
```

Comment:
the var should be declared before

```suggestion
        if (null === $claimedId) {
```